subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/internal/TaskConfiguration.java

```diff
@@ -76,13 +82,24 @@ public boolean isSatisfiedBy(Task element) {
         init.dependsOn(new Callable<String>() {
             @Override
             public String call() throws Exception {
-                if (setupCanBeSkipped.transform(init.getProject()) == null) {
+                if (setupCanBeSkipped.transform(project) == null) {
                     return "wrapper";
                 } else {
                     return null;
                 }
             }
         });
+
+        project.getGradle().getTaskGraph().whenReady(new Action<TaskExecutionGraph>() {
+            @Override
+            public void execute(TaskExecutionGraph taskGraph) {
+                if (setupCanBeSkipped.transform(project) == null && taskGraph.hasTask(init)) {
+                    Wrapper wrapper = (Wrapper) project.getTasks().getByName("wrapper");
+                    BuildInitBuildScriptDsl dsl = BuildInitBuildScriptDsl.fromName(init.getBuildScriptDsl());
+                    wrapper.setDistributionType(dsl.getWrapperDistributionType());
+                }
```

Comment:
What's actually going on here? I don't understand why you're configuring the wrapper task for the build that's already running. Can you add a comment explaining what is going on and why it's necessary?

Reply:
Paraphrasing the code: this sets the `wrapper` task `distributionType` property to the one required by the `BuildInitBuildScriptDsl`. In practice, Groovy generated builds will use the `-bin` distribution and Kotlin ones will use the `-all`distribution.

Reply:
So when you run the `init` task the `wrapper` task also gets invoked with the `distributionType` based upon the desired language?